### ISSUE TITLE
Rule Proposal: Enforce `is` prefix for Boolean prop names

### ISSUE THREAD

**Issue by MrRawbin** [vuejs/eslint-plugin-vue#1101]
**Please describe what the rule should do:**
Enforce the names of Boolean properties to include the `is` prefix.

Perhaps the rule should be configurable to also allow `are`, `should` etc. as prefixes.

**What category should the rule belong to?**

- [X] Enforces code style
- [ ] Warns about a potential error
- [ ] Suggests an alternate way of doing something
- [ ] Other (please specify:)

**Provide 2-3 code examples that this rule should warn about:**
```vue
  props: {
    loaded: {
      type: Boolean,
      required: true,
    },
  }
```

**Comment by MrRawbin** [comment]
I can contribute code for this rule, but if so I'd like some feedback first.

**Comment by ota-meshi** [comment]
Thank you for suggesting this rule. Sorry for late reply.

I'm not sure if this naming rule is general.
I feel that this is often not the case, at least in HTML.
e.g. `disabled`, `readonly`, `draggable`, `contenteditable` and more.

https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes

**Comment by ota-meshi** [comment]
I close this issue because I don't think this rule is probably a general rule.

It would be nice to have a rule that enforces a user's favorite naming convention, but I don't think it's easy to determine its specifications.
